File: snowplow-tracker/src/main/java/com/snowplowanalytics/snowplow/internal/gdpr/GdprConfigurationInterface.java
package com.snowplowanalytics.snowplow.internal.gdpr;

import androidx.annotation.Nullable;

import com.snowplowanalytics.snowplow.util.Basis;

public interface GdprConfigurationInterface {

    /** Basis for processing. */
    @Nullable
    Basis getBasisForProcessing();

    /** ID of a GDPR basis document. */
    @Nullable
    String getDocumentId();

    /** Version of the document. */
    @Nullable
    String getDocumentVersion();

    /** Description of the document. */
    @Nullable
    String getDocumentDescription();
}

File: snowplow-tracker/src/main/java/com/snowplowanalytics/core/gdpr/GdprConfigurationInterface.kt
package com.snowplowanalytics.core.gdpr

import com.snowplowanalytics.snowplow.util.Basis

interface GdprConfigurationInterface {
    /** Basis for processing.  */
    val basisForProcessing: Basis?

    /** ID of a GDPR basis document.  */
    val documentId: String?

    /** Version of the document.  */
    val documentVersion: String?

    /** Description of the document.  */
    val documentDescription: String?
}
